### PR TITLE
Some small event bus fix

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

**Comment by Arsnael** [comment]
Seems breaking some tests in `RabbitMQEventBusTest` : 
```
[561d4d53ac043ac27c9ba92f8637ee322a0ee81b] [ERROR] Errors: 
[561d4d53ac043ac27c9ba92f8637ee322a0ee81b] [ERROR]   RabbitMQEventBusTest$AtLeastOnceTest.inProcessingEventShouldBeReDispatchedToAnotherEventBusWhenOneIsDown:213 » ConditionTimeout
[561d4d53ac043ac27c9ba92f8637ee322a0ee81b] [ERROR]   EventBusConcurrentTestContract$MultiEventBusConcurrentContract.concurrentDispatchGroupShouldDeliverAllEventsToListenersWithMultipleEventBus:167 » ConditionTimeout
[561d4d53ac043ac27c9ba92f8637ee322a0ee81b] [ERROR]   EventBusConcurrentTestContract$SingleEventBusConcurrentContract.concurrentDispatchGroupShouldDeliverAllEventsToListenersWithSingleEventBus:82 » ConditionTimeout
[561d4d53ac043ac27c9ba92f8637ee322a0ee81b] [ERROR]   ErrorHandlingContract.deadLetterShouldNotStoreWhenFailsLessThanMaxRetries:235 » ConditionTimeout
[561d4d53ac043ac27c9ba92f8637ee322a0ee81b] [ERROR]   ErrorHandlingContract.deadLetterShouldStoreWhenFailsGreaterThanMaxRetries:255 » ConditionTimeout
[561d4d53ac043ac27c9ba92f8637ee322a0ee81b] [ERROR]   GroupContract$SingleEventBusGroupContract.groupDeliveryShouldNotExceedRate:89 » ConditionTimeout
[561d4d53ac043ac27c9ba92f8637ee322a0ee81b] [ERROR]   ErrorHandlingContract.listenerShouldReceiveWhenFailsEqualsMaxRetries:115 » ConditionTimeout
[561d4d53ac043ac27c9ba92f8637ee322a0ee81b] [ERROR]   ErrorHandlingContract.listenerShouldReceiveWhenFailsLessThanMaxRetries:98 » ConditionTimeout
[561d4d53ac043ac27c9ba92f8637ee322a0ee81b] [ERROR]   ErrorHandlingContract.retriesBackOffShouldDelayByExponentialGrowth:160->lambda$retriesBackOffShouldDelayByExponentialGrowth$2:168 » IndexOutOfBounds
[561d4d53ac043ac27c9ba92f8637ee322a0ee81b] [ERROR]   ErrorHandlingContract.retryingListenerCallingDispatchShouldNotFail:197 » ConditionTimeout
[561d4d53ac043ac27c9ba92f8637ee322a0ee81b] [INFO] 
[561d4d53ac043ac27c9ba92f8637ee322a0ee81b] [ERROR] Tests run: 118, Failures: 9, Errors: 10, Skipped: 1
```
Please investigate !

**Comment by mbaechler** [comment]
test this please

**Comment by blackheaven** [comment]
test this please

**Comment by blackheaven** [comment]
```
EventBusConcurrentTestContract$SingleEventBusConcurrentContract.concurrentDispatchGroupShouldDeliverAllEventsToListenersWithSingleEventBus:82 » ConditionTimeout
[daa12a0840a9d44815026cd3b3b11264a3961a04] [ERROR]   RabbitMQEventBusTest$LifeCycleTest$SingleEventBus.dispatchShouldWorkAfterNetworkIssuesForNewRegistration:471 » ConditionTimeout
[daa12a0840a9d44815026cd3b3b11264a3961a04] [ERROR]   RabbitMQEventBusTest$LifeCycleTest$SingleEventBus.dispatchShouldWorkAfterRestartForNewRegistration:403 » ConditionTimeout
[daa12a0840a9d44815026cd3b3b11264a3961a04] [ERROR]   RabbitMQEventBusTest$LifeCycleTest$SingleEventBus.redeliverShouldWorkAfterNetworkIssuesForNewRegistration:488 » ConditionTimeout
[daa12a0840a9d44815026cd3b3b11264a3961a04] [ERROR]   RabbitMQEventBusTest$LifeCycleTest$SingleEventBus.redeliverShouldWorkAfterRestartForNewRegistration:429 » ConditionTimeout
[daa12a0840a9d44815026cd3b3b11264a3961a04] [ERROR]   RabbitMQEventBusTest$LifeCycleTest$SingleEventBus.redeliverShouldWorkAfterRestartForOldRegistration:416 » ConditionTimeout
[daa12a0840a9d44815026cd3b3b11264a3961a04] [ERROR]   ErrorHandlingContract.deadLetterShouldNotStoreWhenFailsLessThanMaxRetries:240 » ConditionTimeout
[daa12a0840a9d44815026cd3b3b11264a3961a04] [ERROR]   ErrorHandlingContract.deadLetterShouldStoreWhenDispatchFailsGreaterThanMaxRetries:260 » ConditionTimeout
[daa12a0840a9d44815026cd3b3b11264a3961a04] [ERROR]   ErrorHandlingContract.deadLetterShouldStoreWhenRedeliverFailsGreaterThanMaxRetries:282 » ConditionTimeout
[daa12a0840a9d44815026cd3b3b11264a3961a04] [ERROR]   ErrorHandlingContract.listenerShouldReceiveWhenFailsLessThanMaxRetries:103 » ConditionTimeout
[daa12a0840a9d44815026cd3b3b11264a3961a04] [ERROR]   ErrorHandlingContract.redeliverShouldNotSendEventsToKeyListeners:300 » ConditionTimeout
[daa12a0840a9d44815026cd3b3b11264a3961a04] [ERROR]   ErrorHandlingContract.retryingListenerCallingDispatchShouldNotFail:202 » ConditionTimeout
```

test this please

**Comment by blackheaven** [comment]
test this please

File: mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/GroupRegistration.java [hunk 1]
```diff
@@ -106,8 +106,8 @@ GroupRegistration start() {
         receiverSubscriber = Optional
             .of(createGroupWorkQueue()
                 .then(retryHandler.createRetryExchange(queueName))
-                .thenEmpty(this.consumeWorkQueue())
-                .subscribe());
+                .then(Mono.fromCallable(() -> this.consumeWorkQueue()))
```
Comment (blackheaven) [inline mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/GroupRegistration.java]: @mbaechler is it good for you?

**Comment by blackheaven** [comment]
test this please

File: mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/GroupRegistration.java [hunk 2]
```diff
@@ -125,11 +125,12 @@ GroupRegistration start() {
             .then();
     }
 
-    private Flux<Void> consumeWorkQueue() {
-         return receiver.consumeManualAck(queueName.asString(), new ConsumeOptions().qos(EventBus.EXECUTION_RATE))
+    private Disposable consumeWorkQueue() {
```
Comment (mbaechler) [inline mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/GroupRegistration.java]: What are you doing with this Disposable?

File: mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/GroupRegistration.java (hunk 1 above)
Comment (mbaechler) [reply mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/GroupRegistration.java]: see https://github.com/linagora/james-project/pull/2173#discussion_r298920500

File: mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/GroupRegistration.java (hunk 2 above)
Comment (blackheaven) [reply mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/GroupRegistration.java]: It is stored in `receiverSubscriber`, as in the first version.

The issue with your version is that went from a `Disposable` on a `Flux<Delivery>` and `deliver` (originated in `receiver.consumeManualAck`), to a `Disposable` on a `GroupWorkQueue` creation.

**Comment by mbaechler** [comment]
I can't :+1: on my own PR, but it looks good to me

**Comment by blackheaven** [comment]
```

[3d51d42828cf1f0611c80d1fede425397077c802] Expecting:
[3d51d42828cf1f0611c80d1fede425397077c802]   <2019-07-01T09:06:41.893Z>
[3d51d42828cf1f0611c80d1fede425397077c802] to be after or equals to:
[3d51d42828cf1f0611c80d1fede425397077c802]   <2019-07-01T09:06:41.925Z>
[3d51d42828cf1f0611c80d1fede425397077c802] at ErrorHandlingContract.lambda$retriesBackOffShouldDelayByExponentialGrowth$2(ErrorHandlingContract.java:174)
[3d51d42828cf1f0611c80d1fede425397077c802] 
[3d51d42828cf1f0611c80d1fede425397077c802] 	at org.apache.james.mailbox.events.ErrorHandlingContract.retriesBackOffShouldDelayByExponentialGrowth(ErrorHandlingContract.java:165)
[3d51d42828cf1f0611c80d1fede425397077c802] 
[3d51d42828cf1f0611c80d1fede425397077c802] [INFO] 
[3d51d42828cf1f0611c80d1fede425397077c802] [INFO] Results:
[3d51d42828cf1f0611c80d1fede425397077c802] [INFO] 
[3d51d42828cf1f0611c80d1fede425397077c802] [ERROR] Failures: 
[3d51d42828cf1f0611c80d1fede425397077c802] [ERROR]   ErrorHandlingContract.retriesBackOffShouldDelayByExponentialGrowth:165 
[3d51d42828cf1f0611c80d1fede425397077c802] The following assertion failed:
[3d51d42828cf1f0611c80d1fede425397077c802] 1) 
[3d51d42828cf1f0611c80d1fede425397077c802] Expecting:
[3d51d42828cf1f0611c80d1fede425397077c802]   <2019-07-01T09:06:41.893Z>
[3d51d42828cf1f0611c80d1fede425397077c802] to be after or equals to:
[3d51d42828cf1f0611c80d1fede425397077c802]   <2019-07-01T09:06:41.925Z>
[3d51d42828cf1f0611c80d1fede425397077c802] at ErrorHandlingContract.lambda$retriesBackOffShouldDelayByExponentialGrowth$2(ErrorHandlingContract.java:174)
```

test this please

**Comment by blackheaven** [comment]
```
[INFO] Apache James :: Third Party :: LinShare ............ FAILURE [03:09 min]
```

test this please

**Comment by blackheaven** [comment]
```

[3d51d42828cf1f0611c80d1fede425397077c802] Expecting:
[3d51d42828cf1f0611c80d1fede425397077c802]   <2019-07-02T06:03:14.433Z>
[3d51d42828cf1f0611c80d1fede425397077c802] to be after or equals to:
[3d51d42828cf1f0611c80d1fede425397077c802]   <2019-07-02T06:03:14.450Z>
[3d51d42828cf1f0611c80d1fede425397077c802] at ErrorHandlingContract.lambda$retriesBackOffShouldDelayByExponentialGrowth$2(ErrorHandlingContract.java:174)
[3d51d42828cf1f0611c80d1fede425397077c802] 
[3d51d42828cf1f0611c80d1fede425397077c802] 	at org.apache.james.mailbox.events.ErrorHandlingContract.retriesBackOffShouldDelayByExponentialGrowth(ErrorHandlingContract.java:165)
```

test this please

**Comment by chibenwa** [comment]
Is that related?

**Comment by blackheaven** [comment]
it look likes to be

**Pull request merged**